trieste/models/gpflux/models.py

```diff
@@ -143,10 +144,23 @@ def optimize(self, dataset: Dataset) -> None:
         Optimize the model with the specified `dataset`.
         :param dataset: The data with which to optimize the `model`.
         """
-        self.model_keras.fit(
-            {"inputs": dataset.query_points, "targets": dataset.observations}, **self._fit_args
+        fit_args = dict(self._fit_args)
+
+        # Tell optimizer how many epochs have been used before: the optimizer will "continue"
+        # optimization across multiple BO iterations rather than start fresh at each iteration.
+        # This allows us to monitor training across iterations.
+
+        if "epochs" in fit_args:
+            fit_args["epochs"] = fit_args["epochs"] + self.absolute_epochs
```

Comment:
this would increase the number of epochs in each iteration, if you dont use any stopping rule in callbacks that could quickly become a long training regime, no?